From 5572f20441ef03d5beeab1110a0a96c82f775a31 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E7=82=BD=E5=A4=A9?= <hanxin.hx@alibaba-inc.com>
Date: Tue, 29 Dec 2020 16:12:29 +0800
Subject: [PATCH 35/42] crypto: send-pack: send unencrypted pack to
 receive-pack

When running "git push", add "--no-pack-enc" option to git-pack-objects
to create unencrypted packfile to remote git-receive-pack.

Signed-off-by: Han Xin <hanxin.hx@alibaba-inc.com>
Signed-off-by: Jiang Xin <zhiyou.jx@alibaba-inc.com>
---
 builtin/pack-objects.c    | 2 ++
 send-pack.c               | 1 +
 t/t0940/test-1006-push.sh | 4 ++--
 3 files changed, 5 insertions(+), 2 deletions(-)

diff --git a/builtin/pack-objects.c b/builtin/pack-objects.c
index b7cbb6b468..25ee2aeb75 100644
--- a/builtin/pack-objects.c
+++ b/builtin/pack-objects.c
@@ -4078,6 +4078,8 @@ int cmd_pack_objects(int argc, const char **argv, const char *prefix)
 			 N_("do not pack objects in promisor packfiles")),
 		OPT_BOOL(0, "delta-islands", &use_delta_islands,
 			 N_("respect islands during delta compression")),
+		OPT_BOOL(0, "pack-enc", &agit_crypto_enabled,
+			 N_("encrypt packfile")),
 		OPT_STRING_LIST(0, "uri-protocol", &uri_protocols,
 				N_("protocol"),
 				N_("exclude any configured uploadpack.blobpackfileuri with this protocol")),
diff --git a/send-pack.c b/send-pack.c
index bc0fcdbb00..f0ca2c72c2 100644
--- a/send-pack.c
+++ b/send-pack.c
@@ -74,6 +74,7 @@ static int pack_objects(int fd, struct ref *refs, struct oid_array *advertised,
 	strvec_push(&po.args, "--all-progress-implied");
 	strvec_push(&po.args, "--revs");
 	strvec_push(&po.args, "--stdout");
+	strvec_push(&po.args, "--no-pack-enc");
 	if (args->use_thin_pack)
 		strvec_push(&po.args, "--thin");
 	if (args->use_ofs_delta)
diff --git a/t/t0940/test-1006-push.sh b/t/t0940/test-1006-push.sh
index 0e28da2ce9..dd6bbad44e 100644
--- a/t/t0940/test-1006-push.sh
+++ b/t/t0940/test-1006-push.sh
@@ -2,7 +2,7 @@
 
 # Test crypto on "git-push"
 
-test_expect_failure 'push encrypted repo to normal repo' '
+test_expect_success 'push encrypted repo to normal repo' '
 	cp -R "$COMMON_GITDIR" encrypt.git &&
 	create_bare_repo normal.git &&
 	(
@@ -15,7 +15,7 @@ test_expect_success 'run fsck on normal repo' '
 	git -C normal.git fsck
 '
 
-test_expect_failure 'check log of main' '
+test_expect_success 'check log of main' '
 	git -C normal.git log --oneline |
 		make_user_friendly_and_stable_output >actual &&
 	cat >expect <<-EOF &&
-- 
2.34.1.51.g7e1f4e9345.agit.6.5.6

